rest/rest.go

```diff
@@ -7,150 +7,323 @@ import (
 	"strconv"
 	"strings"
 
+	clienttx "github.com/cosmos/cosmos-sdk/client/tx"
+	codectypes "github.com/cosmos/cosmos-sdk/codec/types"
+	"github.com/cosmos/cosmos-sdk/types"
+	"github.com/cosmos/cosmos-sdk/types/query"
+	"github.com/cosmos/cosmos-sdk/types/tx"
+	"github.com/cosmos/cosmos-sdk/x/auth/legacy/legacytx"
+	"github.com/cosmos/cosmos-sdk/x/auth/signing"
 	"github.com/gin-gonic/gin"
+	"github.com/jackc/pgtype"
 	"github.com/jackc/pgx/v4/pgxpool"
+	"github.com/likecoin/likechain/app"
 	"github.com/likecoin/likecoin-chain-tx-indexer/db"
 	"github.com/likecoin/likecoin-chain-tx-indexer/logger"
+	"github.com/likecoin/likecoin-chain-tx-indexer/util"
 )
 
-type Attribute struct {
-	Type  string
-	Key   string
-	Value string
+var encodingConfig = app.MakeEncodingConfig()
+
+func getUint(query url.Values, key string) (uint64, error) {
+	valueStr := query.Get(key)
+	if valueStr == "" {
+		return 0, nil
+	} else {
+		return strconv.ParseUint(valueStr, 10, 64)
+	}
 }
 
-func getAttributes(query url.Values) (attrs []Attribute) {
+func getPage(query url.Values) (uint64, error) {
+	page, err := getUint(query, "page")
```

Comment:
It seems better to separate the 2 APIs into separate files, so these helper functions won't mix together (which is confusing as I thought it should be `pagination.page` at the first glance).

So I suggest to have at least 3 files: one for legacy Amino endpoint, one for Stargate endpoint, one for common functionalities.

Reply:
Got it, restructured, and ready for review.